screenpy/actor.py

```diff
@@ -101,9 +101,11 @@ def perform(self, action: Performable) -> None:
 
     def cleans_up(self) -> None:
         """Perform any scheduled clean-up tasks."""
-        for task in self.cleanup_tasks:
-            self.perform(task)
-        self.cleanup_tasks = []
+        try:
+            for task in self.cleanup_tasks:
+                self.perform(task)
```

Comment:
This is good, but the issue I can see is if the 1st task fails then none of the rest will be attempted. Is that something we want the library to handle or leave it to the tester to write a custom action to handle?

Reply:
I'm thinking that if the first task the Actor needs to perform as part of cleanup fails, then the rest of the actions will probably also fail. At least, that's how we've been using `has_cleanup_tasks`.

Maybe there's someone else out there who is using them as a to-do list of cleanup actions that can be performed independently, like a series of API requests to delete data?

Reply:
I struggled with this because _we_ can't know what or why their cleanup tasks might fail.  We also can't know if it is wise to continue trying to clean up.  My first inclination is to leave it as is and let the author write their own action to continue trying the remainder of the tasks.  That might be a bit of work for them every time they create an author.

On the other hand, it might be handy for the author to declare how cleanup should go...
"stop cleanup tasks at first failure" vs "keep performing cleanup even if tasks fail"

Reply:
I agree with @bandophahita. "cleanup" is a broad and powerful tool, better to leave it open to the user.

Reply:
Brainstorming scenarios:

### Scenario 1
I need the actor to do the following tasks in any order. These tasks are independent of each other.  The most important task is the database row gets deleted.  If writing the log entry or deleting temp files raises an exception, I don't care, just keep going.

- delete some temp files
- write a log entry 
- remove database row

### Scenario 2
I need cleanup task to occur in a specific order.  Some are dependent on the previous task having been completed. Step 2 really can't be done without logging out properly.  If that were to fail we should stop immediately.

1. logout
2. log the cookie condition after logout
3. delete cache

(ok, maybe a bit contrived, but I think it gets the idea across.) 
The more I consider this problem the more I'm convinced the author needs to handle this on their end.  We can't know which steps should be executed without them providing that context.

Reply:
Scenario 2 is how we currently use cleanup tasks in our suite.

However, Scenario 1 is not satisfied by the existing code, and it is not affected either by the changes in this PR. We'd have to change our approach to try each task in the list, catch any exceptions, and proceed to the next task if one failed.

We can easily do this by adding an `except` line to the try/finally block that checks... i don't know, a setting? Or maybe a property on the Actor? Something else? And if that condition is `False`, it reraises the exception; if it's `True` it just removes the failing task from the cleanup list and continues.

Where should we put that condition?

Reply:
Given the style of screenpy I think it would make the most sense to let the author call an additional method when they are creating the actor.
```python
actor = Actor.named("George").has_cleanup_tasks(t1, t2, t3).whose_cleanup_tasks_should_exit_upon_excpetion()
#or perhaps this method could ALSO accept the tasks
actor = Actor.named("George").has_cleanup_tasks_that_should_exit_upon_excpetion(t1, t2, t3)
# vs 
actor = Actor.named("George").has_cleanup_tasks_that_should_keep_going_at_first_sign_of_trouble(t1, t2, t3)

Reply:
Maybe `Actor.has_dependent_cleanup_tasks` (an alias for `has_cleanup_tasks`?) and `Actor.has_independant_cleanup_tasks`, which set a flag?

Reply:
Although that does open the door to having some dependent tasks and some independent tasks mixed together, maybe they're different lists?

Reply:
I support `dependant` / `independant` cleanup tasks though as you say if a user needs both then it begs the question of which sequence gets executed first. In which case its yet another point at which maybe it makes sense to yield all such handling to the user.

Reply:
Yeah, i realized the same dilemma. I have it set up to do the independent tasks first, then the ordered tasks. That seems like it makes sense.

Reply:
Someone check my logic on this one.  I'm doubting myself. 
I started to rebuttal that order matters...  but it got me questioning: is it possible for order to matter for independent tasks?   If order mattered to that task, doesn't that mean it's not independent?

Reply:
I'm treating "independent tasks" as a set of tasks that don't care about order and don't care if the others succeed. That being said, they _are_ attempted in the order they're given, so you could expect them to be performed in that order.

That's also why i chose the word "ordered" for the other method. If you need one task to complete successfully before another one, you should be using that option.